### PR TITLE
[sui-framework] Reserve addresses for the deny list

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

Added a list of well-known package and object addresses, under the constant `RESERVED`. It would be meaningless to add them to the deny list; as such, they are reserved for internal use--specifically as sentinel values. 

## Test plan 

- Added tests 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by lxfind — APPROVED** [review]

File: crates/sui-framework/packages/sui-framework/sources/deny_list.move [hunk 1]
```diff
@@ -13,10 +13,37 @@ module sui::deny_list {
...
+        @0x7,
+        @0x8,
+        @0x9,
+        @0xA,
+        @0xB,
+        @0xC,
+        @0xD,
+        @0xE,
+        @0xF,
+        @0x403,
```
Comment (tzakian) [inline crates/sui-framework/packages/sui-framework/sources/deny_list.move:42]: bridge?

**Review by tzakian — APPROVED** [review]

File: crates/sui-framework/packages/sui-framework/sources/deny_list.move [hunk 2]
```diff
@@ -13,10 +13,37 @@ module sui::deny_list {
     const ENotSystemAddress: u64 = 0;
     /// The specified address to be removed is not already in the deny list.
     const ENotDenied: u64 = 1;
+    /// The specified address is cannot be added to the deny list.
```
Comment (awelc) [inline crates/sui-framework/packages/sui-framework/sources/deny_list.move:16]:
```suggestion
    /// The specified address cannot be added to the deny list.
```

**Review by awelc — APPROVED** [review]
LGTM other than the typo

**Review by damirka — APPROVED** [review]

File: crates/sui-framework/packages/sui-framework/sources/deny_list.move (hunk 1 above)
Comment (dariorussi) [reply crates/sui-framework/packages/sui-framework/sources/deny_list.move:42]: Bridge package (@patrickkuo)
`const SUI_BRIDGE_ID: address = @0x9;`

Comment (patrickkuo) [reply crates/sui-framework/packages/sui-framework/sources/deny_list.move:42]: bridge object is 0x9 and bridge package is 0xb

File: crates/sui-framework/packages/sui-framework/sources/deny_list.move [hunk 3]
```diff
@@ -13,10 +13,37 @@ module sui::deny_list {
...
+    const EInvalidAddress: u64 = 1;
 
     /// The index into the deny list vector for the `sui::coin::Coin` type.
     const COIN_INDEX: u64 = 0;
 
+    /// These addresses are reserved and cannot be added to the deny list.
+    /// The addresses listed are well known package and object addresses. So it would be
+    /// meaningless to add them to the deny list. As such, they are reserved for internal use.
```
Comment (ronny-mysten) [inline crates/sui-framework/packages/sui-framework/sources/deny_list.move:24]:
```suggestion
    /// meaningless to add them to the deny list.
```

**Pull request merged**